examples/proposed/lock.w

```diff
@@ -0,0 +1,83 @@
+bring cloud;
+
+/**
+ * A lock that can only be acquired by one inflight at a time.
+ *
+ * Internally, this uses a cloud.Counter to keep track of a number.
+ * If the number is 0, the lock is free. If the number is 1, the lock
+ * is acquired.
+ *
+ * This is a very simple lock, and it has a number of limitations:
+  * - It is not reentrant. If a single inflight tries to acquire the lock
+  *   twice, it may block forever.
+  * - It is not fair. If multiple inflights are waiting to acquire the lock,
+  *   there is no guarantee that they will acquire the lock in the order
+  *   that they requested it.
```

Comment:
tbh I would not expect a lock to be fair. 
If I needed a fair lock I would be okay with requiring a `FairLock` resource or something since there's gonna have to be overhead.

Reply:
Fair enough (ha). I've seen in Java and some other languages, locks sometimes have a way to configure fairness so e.g. under contention, the lock is always given to the longest-awaited thread. Maybe it's not something you always expect as a developer, but this kind of feature can help avoid some forms of deadlock I think.

Reply:
To be clear, I think being able to configure fairness is great and I agree with what you're saying, I just meant as a default for something simply called `Lock` I think it's okay if it's unfair.